.github/workflows/master.yml

```diff
@@ -0,0 +1,145 @@
+name: master
+
+on:
+  push:
+    branches: 
+      - master
+  pull_request:
+    branches: 
+      - master
+
+jobs:
+  python3_5:
+    runs-on: ubuntu-latest
+    env:
+      SPARK_VERSION: 2.3.4
+      PANDAS_VERSION: 0.23.4
+      PYARROW_VERSION: 0.10.0
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+
+    - name: setup_python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.5
+
+    - name: before_install
+      run: ./dev/download_travis_dependencies.sh
+
+    - name: install
+      run: |
+        sudo apt-get update
+        pip install --upgrade cython
+        pip install -r requirements-dev.txt
+        pip install pandas==$PANDAS_VERSION pyarrow==$PYARROW_VERSION
+        pip install 'pyspark>=2.4'
+        pip list
+
+    - name: before_script
+      run: |
+        export SPARK_HOME="$HOME/.cache/spark-versions/spark-$SPARK_VERSION-bin-hadoop2.7"
```

Comment:
I think you can move those environment variables to `env`

Reply:
it looks better. :)